openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/core/CoreMatrixAlgorithm.java

```diff
@@ -0,0 +1,538 @@
+
+        algorithm.setEdgeFilter(edgeFilterSequence);
+        algorithm.setTreeEntrySize(this.treeEntrySize);
+        algorithm.setHasTurnWeighting(this.hasTurnWeighting);
+        algorithm.setMaxVisitedNodes(this.maxVisitedNodes);
+        algorithm.setVisitedNodes(this.visitedNodes);
+        algorithm.setTurnWeighting(this.turnWeighting);
+        algorithm.setTargetGraphExplorer(targetGraph.createExplorer());
+        algorithm.setTargetMap(this.targetMap);
+        algorithm.setTargetSet(this.targetSet);
+        algorithm.setSwap(this.swap);
+
+        int[] entryPoints = coreEntryPoints.toArray();
+        int[] exitPoints = coreExitPoints.toArray();
+        algorithm.calcPaths(entryPoints, exitPoints);
+    }
+
+    /**
+     * /
+     * /
+     * __________UTIL
+     * /
+     * /
+     **/
+
+    private boolean isValid(MatrixLocations srcData, MatrixLocations dstData) {
+        return !(!srcData.hasValidNodes() || !dstData.hasValidNodes());
+    }
+
+    /**
+     * Search is more efficient for low source count and high destination count than the other way around.
+     * If there are more sources than destinations, they get swapped and all calculations are done backwards.
+     * The final result gets unswapped to return correct results.
+     * @param srcData original Source data
+     * @param dstData original Destination data
+     * @return
+     */
+    private boolean checkSwapSrcDst(MatrixLocations srcData, MatrixLocations dstData){
+        return(srcData.size() > dstData.size());
+    }
+
+    /**
+     * Invert the results matrix (represented by flattened array) in case src and dst were swapped
+     * @param srcData the original unswapped source data
+     * @param dstData the original unswapped destination data
+     * @param times the swapped array of results
+     * @param distances the swapped array of results
+     * @param weights the swapped array of results
+     * @return array of unswapped result arrays [times, distances, weights]
+     */
+    private float[][] swapResults(MatrixLocations srcData, MatrixLocations dstData, float[] times, float[] distances, float[] weights) {
+        boolean hasTimes = times != null;
+        boolean hasDistances = distances != null;
+        boolean hasWeights = weights != null;
+        float[] newTimes = new float[0];
+        float[] newDistances = new float[0];
+        float[] newWeights = new float[0];
+        if(hasTimes)
+            newTimes = new float[times.length];
+        if(hasDistances)
+            newDistances = new float[distances.length];
+        if(hasWeights)
+            newWeights = new float[weights.length];
+
+        int i = 0;
+        int srcSize = srcData.size();
+        int dstSize = dstData.size();
+        for (int dst = 0; dst < dstSize; dst++){
+            for(int src = 0; src < srcSize; src++){
+                if(hasTimes)
+                    newTimes[dst + src * dstSize] = times[i];
```

Comment:
It might be actually smart to introduce an intermediate variable holding the common index `dst + src * dstSize`.

Reply:
done